src/csvwidget/widget.ts

```diff
@@ -217,6 +217,27 @@ function createDelimiterSwitcherNode(): HTMLElement {
 export
 class CSVWidgetFactory extends ABCWidgetFactory<CSVWidget, DocumentRegistry.IModel> {
   /**
+   * The name of the widget to display in dialogs.
+   */
+  get name(): string {
+    return 'Table';
+  }
+
+  /**
+   * The file extensions the widget can view.
+   */
+  get fileExtensions(): string[] {
+    return ['.csv'];
```

Comment:
Ah, I just realized that even the name wasn't part of the factory itself before.  Still, I think it makes sense to have the factory have default values for all of these options but allow overrides.